src/filters/compression.rs

```diff
@@ -56,7 +58,10 @@ pub struct Compression<F> {
 /// ```
 pub fn gzip() -> Compression<impl Fn(CompressionProps) -> Response + Copy> {
     let func = move |mut props: CompressionProps| {
-        let body = Body::wrap_stream(GzipEncoder::new(props.body));
+        let body = Body::wrap_stream(FramedRead::new(
```

Comment:
why did you prefer `FramedRead` over `ReaderStream`?

Reply:
@jxs To be honest, I don't remember exactly, but it was probably from an example. Do you think ReaderStream is preferable?

Reply:
It was inspired from the reqwest implementation https://github.com/seanmonstar/reqwest/blob/a19eb3419637856d6d42f55dff6e4790613b2933/src/async_impl/decoder.rs#L47

Reply:
Thanks @paolobarbolini, I can't bring myself to remember where I found this solution.

Reply:
thanks @paolobarbolini. @aknuds1 I asked because I was not sure either, Previously `FrameReader` was the way to achieve `AsyncRead` to `Stream` conversion, but since `ReaderStream` [was introduced](https://github.com/tokio-rs/tokio/issues/2705) I would maybe suggest `ReaderStream` as it's simpler, and recommended for this cases on [tokio-util doc](https://docs.rs/tokio-stream/0.1.1/tokio_stream/#conversion-to-and-from-asyncreadasyncwrite)

Reply:
@seanmonstar Do you have any opinion on this?

Reply:
I converted from FramedRead into ReaderStream, ass @jxs recommended. I don't really know personally if there are any practical differences, but I can tell that the API is simpler (a second argument to `FramedRead::new` can be dropped). I trust @jxs knows more about this than I do, and I'll let @seanmonstar be the judge.